src/main/java/ludum/dare/actors/player/RunState.java

```diff
@@ -0,0 +1,30 @@
+package ludum.dare.actors.player;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Input;
+import ludum.dare.interfaces.IComponent;
+import ludum.dare.interfaces.IState;
+
+import java.util.Set;
+
+public class RunState extends IState {
+
+    public RunState(Set<IComponent> components) {
+        super(components);
+    }
+
+    public void enter() {
+        // TODO: switch to a running animation
+    }
+
+    public void exit() {
+
+    }
+
+    public IState update(float delta) {
+        // TODO: switch the direction of the animation based on something
+
+        if (!Gdx.input.isKeyPressed(Input.Keys.LEFT)) return new StandState(components);
```

Comment:
Also --- I think my confusion is coming from this compared to the PlayerInputController class in Jump. It's function is basically to handle how input is handled by a physics body. Are we intended for certain components to be activated/deactivated depending on the state of the GameObject?